src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.SymbolAndSelectionInfo.cs

```diff
@@ -9,15 +9,13 @@ namespace Microsoft.CodeAnalysis.Completion.Providers
 {
     internal abstract partial class AbstractSymbolCompletionProvider<TSyntaxContext>
     {
-        private sealed class CompletionLinkedFilesSymbolEquivalenceComparer : IEqualityComparer<(ISymbol symbol, bool preselect)>
+        protected readonly record struct SymbolAndSelectionInfo(ISymbol Symbol, bool Preselect)
         {
-            public static readonly CompletionLinkedFilesSymbolEquivalenceComparer Instance = new();
+            public bool Equals(SymbolAndSelectionInfo other)
+                => LinkedFilesSymbolEquivalenceComparer.Instance.Equals(Symbol, other.Symbol);
 
-            public bool Equals((ISymbol symbol, bool preselect) x, (ISymbol symbol, bool preselect) y)
-                => LinkedFilesSymbolEquivalenceComparer.Instance.Equals(x.symbol, y.symbol);
-
-            public int GetHashCode((ISymbol symbol, bool preselect) obj)
-                => LinkedFilesSymbolEquivalenceComparer.Instance.GetHashCode(obj.symbol);
+            public override int GetHashCode() 
+                => LinkedFilesSymbolEquivalenceComparer.Instance.GetHashCode(Symbol);
```

Comment:
consider doc'ing that our intentional semantics here are that things are equivalent just basedon symbol.